### PR TITLE
Adding TAR and gunzip support

### LINKED ISSUE #541: Utility function to handle tar files
There are times when the data files are delivered in a TAR file. The DQ Office have some utilities for handling TAR data files, including creation of a temporary directory and clean up after the extracted files are used. Would be good to add these utility functions to ACT for others to utilize.

### PR DESCRIPTION
**Pull request by kenkehoe**
This adds in some functions to create and extract from TAR/gunzip files. Also updated the netCDF reader to just work with TAR and gunzip files. Resolves #541 .

### CODE REVIEW

**Comment by kenkehoe** [comment]
I am getting a strange error?

FAILED act/tests/test_io.py::test_read_netcdf_tarfiles - NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpe7rvjwup\\uwSmSKsq7lOAEjrf5W6z\\sgpmetE13.b1.20190101.000000.cdf'

Any idea what the issue could be? I am creating a random directory name as part of the process.

**Comment by AdamTheisen** [comment]
@kenkehoe I'm rerunning the tests to make sure but I'm wondering if it has to do with creating a directory outside the package on this runner.  I wonder if creating a temp directory somewhere in the ACT folder structure would have different permissions.

**Comment by AdamTheisen** [comment]
@kenkehoe  I just resolved a conflict with the files so it's running again but I can try and test out on my windows machine later today on the pytest error.

```
FAILED act/tests/test_io.py::test_read_netcdf_tarfiles - PermissionError: [WinError 32] The process cannot access the file 
because it is being used by another process: 
'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpwod4mqnu\\aovgrX0cOy2UjSFAj0vz\\sgpmetE13.b1.20190101.000000.cdf'

FAILED act/tests/test_io.py::test_read_netcdf_gztarfiles - ValueError: did not find a match in any of xarray's currently 
installed IO backends ['netcdf4', 'scipy']. Consider explicitly selecting one of the installed engines via the ``engine`` 
parameter, or installing additional IO dependencies, see:
```

**Comment by AdamTheisen** [comment]
@kenkehoe  I was able to recreate the error on my windows machine and will try and work through it this week as I can.

**Comment by kenkehoe** [comment]
@AdamTheisen let me know how I can help with this. I did a few changes/comment out code to try and help determine where the issue could be.

**Comment by AdamTheisen** [comment]
@kenkehoe I don't have it fully figured out yet but I think it has to do with the circular nature of how you have it set up.  I.e. that in `read_netcdf` you call `check_if_tar_gz_file` where it checks the file and then calls `read_netcdf` once it's unpacked the files.  I think this is complicating things and causing issues with file permissions on windows.  It might be best to have the function unpack the files and return a list of filenames for `read_netcdf` to then proceed with reading.

The other issue was that the files returned from line 178 in `test_read_netcdf_gztarfiles` were tar files which I guess would be untarred if passed back through `read_netcdf` but again, I think it's too many circular loops here.

Thoughts?

**Comment by kenkehoe** [comment]
Yeah, I was thinking that may be the issue. I'll refactor it and see I can make it less circular.

**Comment by AdamTheisen** [comment]
@kenkehoe  it looks like a couple pep8 issues

```
.\act\tests\test_io.py:789:5: E303 too many blank lines (2)
.\act\tests\test_io.py:796:1: W391 blank line at end of file
```

**Comment by kenkehoe** [comment]
@AdamTheisen I don't understand the errors. Can you just rerun?

**Comment by kenkehoe** [comment]
All checks have passed. I just shed a tear!

**Comment by AdamTheisen** [comment]
@kenkehoe my only thought here is if the io_utils.py file should be in the utils directory

**Comment by AdamTheisen** [comment]
@zssherman  or @mgrover1  any thoughts here on the utils file?

**Comment by zssherman** [comment]
I think either or is fine, or the argument can be made for both, it is unpacking and changing files etc. But I would say it's not necessarily reading a file in the aspect of some of the other readers. My vote would be utils

**Comment by mgrover1** [comment]
@AdamTheisen - I think it would be best to move to the `utils` module, to be consistent with the rest of the utils functions. 

@kenkehoe can you move `io_utils.py` over there?

**Comment by kenkehoe** [comment]
OK I'll move the functions to the utils area.

**Comment by zssherman** [comment]
@kenkehoe PEP8 line continuation, and @mgrover1 the readded mmcr code PR will just squash when this is merged correct? Besides that, once test pass, i think this is good to go

**Comment by kenkehoe** [comment]
@zssherman I resolved the merge conflict from MMCR PR in my code. So once the tests pass should be good to go.

**Review by zssherman — APPROVED** [review]

**Comment by zssherman** [comment]
Looks good to me, @AdamTheisen @mgrover1 any last thoughts? Figured a large PR good for one last check

File: act/io/__init__.py [hunk 1]
```diff
@@ -34,5 +35,12 @@
             'read_psl_radar_fmcw_moment',
         ],
         'pysp2': ['read_hk_file', 'read_sp2', 'read_sp2_dat'],
+        'io_utils': ['pack_tar',
```
Comment (AdamTheisen) [inline act/io/__init__.py:38]: @kenkehoe I think we need to move this over to the __init__ in the utils area as well.

**Review by AdamTheisen — CHANGES_REQUESTED** [review]
@kenkehoe @zssherman other than this one change I think it should be good to go.

**Pull request merged**